projects/canopy/src/lib/filter-buttons/filter-button/filter-button.component.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Component,
+  HostBinding,
+  Input,
+  OnInit,
+  Optional,
+  Self,
+  ViewEncapsulation,
+  Host,
+  SkipSelf,
+} from '@angular/core';
+import {
+  NgControl,
+  ControlValueAccessor,
+  FormGroupDirective,
+  ControlContainer,
+} from '@angular/forms';
+
+import { LgErrorStateMatcher } from '../../forms/validation/error-state-matcher';
+import { LgFilterGroupComponent } from '../filter-group.component';
+import { FilterVariant } from '../filter-button.interface';
+
+let uniqueId = 0;
+
+@Component({
+  selector: 'lg-filter-button',
+  templateUrl: './filter-button.component.html',
+  styleUrls: ['./filter-button.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgFilterButtonComponent implements OnInit, ControlValueAccessor {
+  nexUniqueId = ++uniqueId;
+  @Input() checked = false;
+  @Input() id = `lg-filter-button-${this.nexUniqueId}`;
+  @Input() name = `lg-filter-button-${this.nexUniqueId}`;
+  @Input() value: string | boolean;
+  @Input() ariaDescribedBy: string;
+  @Input() variant: FilterVariant;
+
+  @Input()
+  _disabled = false;
+  get disabled(): boolean {
+    if (this.controlArray) {
+      return this._disabled || (this.controlArray !== null && this.controlArray.disabled);
+    } else {
+      return this._disabled || (this.filterGroup !== null && this.filterGroup.disabled);
+    }
+  }
+  set disabled(isDisabled: boolean) {
+    this._disabled = isDisabled;
+  }
+
+  @HostBinding('class.lg-filter-button') class = true;
+  @HostBinding('class.lg-filter-button--error') get errorClass() {
+    if (this.variant === 'select-multiple') {
+      if (this.control) {
+        return this.errorState.isControlInvalid(this.control, this.controlContainer);
+      } else {
+        return this.errorState.isControlInvalid(this.controlArray, this.controlContainer);
+      }
+    } else {
+      return this.errorState.isControlInvalid(this.control, this.controlContainer);
+    }
+  }
```

Comment:
This is more of just a style thing, so take it or leave it, but I'm a fan of the early return here as I think it makes it a bit more readable
```suggestion
  if (this.variant === 'select-multiple') {
    if (this.control) {
      return this.errorState.isControlInvalid(this.control, this.controlContainer);
    } 
    return this.errorState.isControlInvalid(this.controlArray, this.controlContainer);
  }
  return this.errorState.isControlInvalid(this.control, this.controlContainer);
}
```